custom_components/unifiprotect/entity.py

```diff
@@ -5,45 +5,45 @@
 from homeassistant.core import callback
 import homeassistant.helpers.device_registry as dr
 from homeassistant.helpers.entity import DeviceInfo, Entity
+from pyunifiprotect import ProtectApiClient
+from pyunifiprotect.data.base import ProtectAdoptableDeviceModel
 
 from .const import ATTR_DEVICE_MODEL, DEFAULT_ATTRIBUTION, DEFAULT_BRAND, DOMAIN
+from .data import UnifiProtectData
 
 
 class UnifiProtectEntity(Entity):
     """Base class for unifi protect entities."""
 
-    def __init__(self, upv_object, protect_data, server_info, device_id, description):
+    def __init__(
+        self,
+        protect: ProtectApiClient,
+        protect_data: UnifiProtectData,
+        device: ProtectAdoptableDeviceModel,
+        description,
+    ):
         """Initialize the entity."""
         super().__init__()
         self._attr_should_poll = False
 
         if description:
             self.entity_description = description
 
-        self.upv_object = upv_object
-        self.protect_data = protect_data
-        self._device_id = device_id
-        self._device_data = self.protect_data.data[self._device_id]
-        self._device_name = self._device_data["name"]
-        self._attr_name = self._device_name
-        self._mac = self._device_data["mac"]
-        self._firmware_version = self._device_data["firmware_version"]
-        self._server_id = server_info["server_id"]
-        self._server_ip = server_info["server_ip"]
-        self._device_type = self._device_data["type"]
-        self._model = self._device_data["model"]
+        self.protect: ProtectApiClient = protect
+        self.protect_data: UnifiProtectData = protect_data
+        self.device: ProtectAdoptableDeviceModel = device
```

Comment:
This is getting set in the child class sometimes.  Can we avoid that?

Reply:
I did that on purpose to reset the type of the object. For when it is a more specific child class (ex: in `camera.py` it is _always_ a camera object). You mentioned now having the class attributes outside of the init before. 

Is there a better way to do that?

Reply:
Always set `self.device` in the child class before calling `super()`